### ISSUE TITLE
Measurements Panel display units of SUV even when PROPCOUNTS is reported [Bug] 

### ISSUE THREAD

**Issue by BrandonDatUHN** [OHIF/Viewers#3439]
### Describe the Bug

For preforming an annotation on a non-attenuation corrected PET image the units of the measurement are not shown if they are an NAC image (units PROPCNTS).  This isn't ideal since for CT the units are shown as HU and for PET AC they show SUV but not a big issue.

Bug is that the measurement panel still shows the units in SUV.

![image](https://github.com/OHIF/Viewers/assets/112980389/d2b2aa6b-883e-4825-b07a-1e270e6f80fb)

### Steps to Reproduce

1. Go to https://v3-demo.ohif.org/viewer?StudyInstanceUIDs=1.2.840.113619.2.290.3.3767434740.226.1600859119.501
2. Select the NAC image
3. Draw a contour of any kind (I picked circle)
4. Mean, max stdev on image show as unitless, measurement panel shows as SUV (though the number displayed is in propcounts)

### The current behavior

Measurement panel shows the units of a measurement on a PET NAC image with a tag SUV 

### The expected behavior

Measurement panel should show the units of a measurement on a PET NAC image with a tag propcounts 

### OS

Win 10 

### Node version

Using OHIF Demo 

### Browser

 Chrome 113.0.0

**Comment by 03shek** [comment]
@jbocce I was asked by @Michael-Andersen to chime in while @BrandonDatUHN is away. Not a nuc med expert here, but as I understand it, there are 2 types of PET images.

Dicom tags involved:
- (0054,1001) Units (variants listing: [https://dicom.innolitics.com/ciods/pet-image/pet-series/00541001](https://dicom.innolitics.com/ciods/pet-image/pet-series/00541001))
- (0028,1052) Rescale Intercept
- (0028,1053) Rescale Slope 
- Tags for conversion to SUV

![image](https://github.com/OHIF/Viewers/assets/115667958/eff48693-6131-44fc-97f1-634ebb1c224b)

- We want to display the right-most image in OHIFv3 and their respective units for both types of PET images. 
- The units stated should match between the viewport and the measurements list.

**Comment by jbocce** [comment]
Thanks so much @03shek. 

So to confirm, the units to show in the viewport and the measurements list should always come from the DICOM correct? 

Should there be a fallback in case the tag value is blank? It says that it is required, but you never know.

**Comment by 03shek** [comment]
> the units to show in the viewport and the measurements list should always come from the DICOM correct?

Yes, for PET. Fallback for PET modality in case you find the tag value is blank could be something like "unitless". You'd want it to be differentiated from raw values somehow.

**Comment by jbocce** [comment]
Thanks @03shek!

So for raw values showing no unit makes sense, correct?

For PET, if there were no rescale slope and intercept applied, would that be considered raw?

**Comment by 03shek** [comment]
Agreed x2

**Comment by sedghi** [comment]
@03shek @salimkanoun 

Just to double check, since it is contradictory to what I thought, if it is PT and it is scaled (rescale and slope) and applied the patient weight the unit should be SUV right?

**Comment by 03shek** [comment]
Only if the units of the PT image is BQML

**Comment by 03shek** [comment]
And just to be clear, along with patient weight, there should be injected dose, or technically called Radionuclide Total Dose (0018,1074), within what I called "Tags for conversion to SUV"

**Comment by salimkanoun** [comment]
Yes weight is not enough to calculate suv, the full calculation of suv should pass to display suv. Otherwise fall back to the unit displayed in the unit tag

**Comment by jbocce** [comment]
Ok, thanks @03shek and @salimkanoun. It sounds like we need a clearer understanding of what the proper SUV calculation is because I believe OHIF is only currently using the patient weight to do the SUV scaling. Perhaps someone could help us flesh out the details of the proper calculation? Thanks.

**Comment by jbocce** [comment]
Thanks @03shek. If you could please have a look at the code snippet below (together with line numbers on the left) and answer a few questions for us then that would be greatly appreciated.  The snippet is basically the various PT scaling that is performed whereby the `array` contains the raw image pixel values and `suvbw` is the SUV body weight factor.

1. So in the first branch of the `if` statement is OHIF's SUV calculation based on body weight. Furthermore, note that this is the only and exiting SUV calculation OHIF does. What should be shown as a unit suffix to that calculated amount on line 3?

2. In the last/else branch of the `if` statement is the application of rescale slope and intercept (i.e. the non SUV case). Could you please confirm that for the calculated amount on line 7 we should show the contents of the `(0054,1001)` tag as the unit suffix?

Thanks again.

```
1  if (scalingParameters.modality === 'PT' && typeof suvbw === 'number') {
2   for (let i = 0; i < arrayLength; i++) {
3      array[i] = suvbw * (array[i] * rescaleSlope + rescaleIntercept);
4    }
5  } else {
6    for (let i = 0; i < arrayLength; i++) {
7      array[i] = array[i] * rescaleSlope + rescaleIntercept;
8    }
9   }

```

**Comment by salimkanoun** [comment]
Hi there, 2y ago I worked with Erik on SUV calcualation.
We use the QIBA pseudo code, the SUV implementation of Fiji and I selected a dozen of cases of differents DICOM reprensenting different calculation cases with gold truth value comming from Fiji and a major PET/CT manufacturer workstation (we analysed our dataset of 6000 PET/CT to find those case that have different calculation rules).

He implemented the calcultation in the calculate-suv librairy https://github.com/cornerstonejs/calculate-suv and used the sample dicom for unit testing.

What you saw at "suvbw" factor is the conversion factor calculated with calculated suv which give the conversion factor to get SUV value, is is calculated from calculted-suv and injected to volume metadata to make the SUV value computation in cornerstone.

I think the design is good for SUV.
However I took the chance to say that SUV is not the only calculation being used in PET/CT especially there is also SUL (lean body mass) which has itself different algorithm
In this pull request https://github.com/cornerstonejs/calculate-suv/pull/18 I add the recomanded method by the EANM.

The not ideal choice is that if you look at cornerstone, all these conversion unit and injected as metadata of the image like the suv conversion factor, so you will have a lot of conversion factors, SUV, SULbsa, SULjanma.
There are also glyceamia corrected SUV that might be needed.

The problem we will have is that the annotation tools are highly coupled, there is one class managing reference & geometry, calculations of metrics and rendering of anotation with special case in the rendering (if PT display SUL, SUV etc...)

I think we will need something more decoupled, it's fine to store the SUV conversion factor but then at OHIF level it would be better to be able to inject calculation and rendering callback so OHIF (or extension or mode) will implement special case for calculations while cornerstone still be limited to rendering a default model for quantification that could be fine tuned in the parent project.

This might introduce significant rewriting in cornerstone, however we are ready to help in this effort,

Best regards,

Salim

**Comment by 03shek** [comment]
@jbocce Re:

1. Line 3 requires the incorporation of the injected dose tag, Radionuclide Total Dose `(0018,1074)`, as shown below. Line 3's result unit suffix should be SUV, or SUVbw if you wish to be extra clear and offer future options such as those suggested by @salimkanoun. 
`3      array[i] = suvbw/injdose * (array[i] * rescaleSlope + rescaleIntercept);`

2. Yes, unit suffix of Line 7's result should be as specified by the Units tag `(0054,1001)`

**Comment by salimkanoun** [comment]
the suvbw factor already includes decay correction and weight normalization (see calculate suv librairy)

**Comment by 03shek** [comment]
Thanks @salimkanoun.

This brings up an interesting point... only SUVbw is decay-corrected according to lines 1-9 posted by @jbocce. My personal opinion is that lines 6-9 should be decay-corrected too. What do you think? According to the description of the Decay Correction tag `(0054,1102)`, it is a required tag for modality PT.

**Comment by salimkanoun** [comment]
No i don't think so, I never seen decay correction value applyed for non SUV unit. 
The point of SUV is that you normalize the mesure uptake in the image to the injected dose 1h before injection so in this case you need to decay correction to make this normalization.

But if you don't normalize to the injected dose i don't see any value to decay correct the raw value. Also you may also face missing data such as missing inject time and you will have additional special case if no SUV => try to decay correct, if not => raw value.... start to be complicated...

**Comment by 03shek** [comment]
Yeah I'm not sure. The only instance where it could be required is dose/scanner calibration purposes I guess. (More frequent an issue for NM SPECT).

Sorry, I'm not a dev and can't wade into the depths of the OHIF or cornerstone code knowledgeably.

At the same time, I want to make sure that even if the viewer cannot calculate SUVbw, and can only display BQML (ie., use the else statement in Lines 1-9 above (?) ), the BQML values would at least be properly decay-corrected.

**Comment by JennyLeePMH** [comment]
```
if modality == PT:
    if Units tag (0054,1001) == "BQML":
        if suvbw exists and is a number:
            # this is a PET AC image, and all necessary dicom tags are presents to compute the suvbw factor
            array[i] = suvbw * (array[i] * rescaleSlope + rescaleIntercept)
            unit_to_display = "SUV"
        else:
            # this is a PET AC images, but the suvbw factor is unavailable because of missing patient weight or something else 
            array[i] = (array[i] * rescaleSlope + rescaleIntercept)
            unit_to_display = "BQML"
    elif Units tag (0054,1001) in ["PROPCNTS", "PROPCPS"]
        # this is a PET NAC image
        array[i] = (array[i] * rescaleSlope + rescaleIntercept)
        unit_to_display = Units tag (0054,1001)

        
```
Does the above pseudo code make sense to everyone?  

Note that Brandon's bug was about the unit displayed being incorrect, the computed number is actually correct.  The above screen captured images is a PET NAC.

**Comment by salimkanoun** [comment]
> ```
> if modality == PT:
>     if Units tag (0054,1001) == "BQML":
>         if suvbw exists and is a number:
>             # this is a PET AC image, and all necessary dicom tags are presents to compute the suvbw factor
>             array[i] = suvbw * (array[i] * rescaleSlope + rescaleIntercept)
>             unit_to_display = "SUV"
>         else:
>             # this is a PET AC images, but the suvbw factor is unavailable because of missing patient weight or something else 
>             array[i] = (array[i] * rescaleSlope + rescaleIntercept)
>             unit_to_display = "BQML"
>     elif Units tag (0054,1001) in ["PROPCNTS", "PROPCPS"]
>         # this is a PET NAC image
>         array[i] = (array[i] * rescaleSlope + rescaleIntercept)
>         unit_to_display = Units tag (0054,1001)
> 
>         
> ```
> Does the above pseudo code make sense to everyone?  
> 
> Note that Brandon's bug was about the unit displayed being incorrect, the computed number is actually correct.  The above screen captured images is a PET NAC.   

Hi, no you shouldn't make suv calculation   depending on bqml units, for instance Philips somethimes provides "count" units with a private tag to convert it in SUV (which is handler in the suv computation librairy)

So the pseudo code should be easier :

```
if modality == PT
   If  ATTN in image type (check AC image) and suvbw exists and is a number:
      # this is a PET AC image, and all necessary dicom tags are presents to compute the suvbw factor
           array[i] = suvbw * (array[i] * rescaleSlope + rescaleIntercept)
         unit_to_display = "SUV"
  else:
            # either  NAC images, or the suvbw factor is unavailable because of missing patient weight or something 
    array[i] = (array[i] * rescaleSlope + rescaleIntercept)
     unit_to_display = Units tag (0054,1001)
        
 ```

**Comment by 03shek** [comment]
Thanks everyone. It does look like `if units == BQML` and the Philips exception is handled explicitly within the `calculate-suv` computation library, even that imageType includes `DECY` in addition to `ATTN`. 

There's a part of me that wants to vouch for preclinical PET researchers who could use decay-corrected BQML when decay-correction is available (to eventually get %ID/g after weighing scanned specimens), but I don't think they're much in OHIF's target demo.

Hopefully @jbocce has everything he needs...

**Comment by jbocce** [comment]
Thanks to everyone on my part as well. Yes I believe I have everything I need. One favour I will ask of @03shek is to please update the flowchart from earlier in the issue to  include the actual desired unit suffix in EACH of the image boxes. In particular, for the SUV Image box to include `unit suffix: SUV`. If that could be done then that would be greatly appreciated. Thanks so much!

**Comment by BrandonDatUHN** [comment]
Thanks.  This specific fix in the measurements panel appears to work now.  However I'm noticing that PET BQML images with no patient weight no longer display at all.  It seems it attempts to convert to SUV (even if there is no weight or weight is set to 0) and then no image is shown.

Here is an example set 

https://fileshare.uhn.ca/download/f74813e9-2522-4357-87e2-ad7115c7fb82  password "animal"

**Comment by BrandonDatUHN** [comment]
Hi @jbocce  Are you able to reproduce the error with PET BQML images and no Body Weight?

**Comment by jbocce** [comment]
Hi @BrandonDatUHN. I just returned from vacation yesterday and yes I am able to reproduce it. However, this problem concerning  PET BQML images and no Body Weight appears to have existed prior to the changes for this issue and thus I believe are unrelated. In fact, I tried it in the released version of OHIF (version 3.6 we are currently working on 3.7) https://viewer.ohif.org/localbasic by uploading the provided study and it appears broken there too.

I think it is best to perhaps open a new issue. Thoughts?
